core/templates/dev/head/pages/story-viewer-page/chapters-list/story-viewer-chapters-list.directive.ts

```diff
@@ -85,26 +87,53 @@ angular.module('oppia').directive('storyViewerChaptersList', [
 
           $anchorScroll.yOffset = -80;
 
-          ctrl.setIconHighlight = function(index) {
-            ctrl.activeHighlightedIconIndex = index;
+          ctrl.isVisibleCard = function(storyNode) {
+            if (!storyNode) {
+              return false;
+            }
+            var isNextPendingNode = (
+              ctrl.storyPlaythroughObject.getNextPendingNodeId() ===
+              storyNode.getId());
+            return storyNode.isCompleted() || isNextPendingNode;
           };
 
-          ctrl.unsetIconHighlight = function() {
-            ctrl.activeHighlightedIconIndex = -1;
+          ctrl.getIconTitle = function(node, isMobile, index) {
+            var prefix = '';
+            if (isMobile) {
+              prefix = (index + 1) + '. ';
+            }
+            if (ctrl.isVisibleCard(node)) {
+              if (node.isCompleted()) {
+                return prefix + node.getTitle() + ' - Completed!';
```

Comment:
Should this text be stored in a constants file somewhere? This seems like it would be hard to translate

Reply:
+1

Reply:
Do you mean just the 'completed' at the end? Since, the rest of it is variable.

Reply:
One thing you might want to do is to make some kind of template string that has the title as an interpolated variable, since the order might be different in different languages.

See e.g. what we do for I18N_FORMS_TYPE_NUMBER_AT_LEAST.

Reply:
Done